main/src/ca/uwaterloo/flix/api/lsp/provider/completion/VarCompleter.scala

```diff
@@ -32,6 +33,20 @@ object VarCompleter extends Completer {
       case Entity.FormalParam(fparam) => Completion.VarCompletion(fparam.sym, fparam.tpe)
     }
 
-    iter.toList
+    iter.toList.filter(comp => matchesVar(comp.sym, context.word))
+  }
+
+  /**
+    * Checks that the varSym matches the prefix
+    *
+    * @param sym  the varSym.
+    * @param word the current prefix.
+    * @return     true, if the var matches prefix, false otherwise.
+    */
+  private def matchesVar(sym: Symbol.VarSym, word: String): Boolean = {
+    if (word.nonEmpty && word.head.isUpper)
```

Comment:
Variables cannot be uppercase (??)

Reply:
I was a bit too quick there. That is indeed true.